huaweicloud/services/acceptance/tms/resource_huaweicloud_tms_resource_tags_test.go

```diff
@@ -0,0 +1,174 @@
+package tms
+
+import (
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk"
+	tags "github.com/chnsz/golangsdk/openstack/tms/v1/resourcetags"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/tms"
+)
+
+func getResourceTagsFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	client, err := cfg.TmsV2Client()
+	if err != nil {
+		return nil, fmt.Errorf("error creating TMS v2 client: %s", err)
+	}
+
+	var (
+		projectId       = state.Primary.Attributes["project_id"]
+		resourcesLen, _ = strconv.Atoi(state.Primary.Attributes["resources.#"])
+		tagsLen, _      = strconv.Atoi(state.Primary.Attributes["tags.%"])
+		tagsConfigured  = false
+	)
+
+	for i := 0; i < resourcesLen; i++ {
+		resourceId := state.Primary.Attributes[fmt.Sprintf("resources.%d.resource_id", i)]
+		opts := tags.QueryOpts{
+			ResourceId:   resourceId,
+			ResourceType: state.Primary.Attributes[fmt.Sprintf("resources.%d.resource_type", i)],
+			ProjectId:    projectId,
+		}
+		resp, err := tags.Get(client, opts)
+		if err != nil {
+			return nil, fmt.Errorf("error query resource (%s) tags: %s", resourceId, err)
+		}
+		actualTags := tms.FlattenTagsToMap(resp)
+		if len(actualTags) != tagsLen {
+			return nil, fmt.Errorf("some tags were not set successfully")
+		}
+		if len(actualTags) > 0 {
+			tagsConfigured = true
+		}
+	}
+	if !tagsConfigured {
+		return nil, golangsdk.ErrDefault404{}
+	}
+	return tagsConfigured, nil
+}
+
+func TestAccResourceTags_basic(t *testing.T) {
+	var (
+		tagsConfigured bool
+
+		rName    = "huaweicloud_tms_resource_tags.test"
+		basicCfg = testAccResourceTags_base()
+		rc       = acceptance.InitResourceCheck(rName, &tagsConfigured, getResourceTagsFunc)
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
```

Comment:
Since `acceptance.HW_PROJECT_ID` is used in this unit test. Better add a pre test for this env field